hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/HazelcastWriters.java

```diff
@@ -55,6 +60,149 @@
+                    IMap map = instance.getMap(name);
+                    Map<K, T> tmpMap = new HashMap<>();
+                    AbstractEntryProcessor entryProcessor = new AbstractEntryProcessor() {
+                        @Override
+                        public Object process(Entry entry) {
+                            V oldValue = (V) entry.getValue();
+                            T item = tmpMap.get(entry.getKey());
+                            V newValue = updateFn.apply(item, oldValue);
+                            entry.setValue(newValue);
+                            return null;
+                        }
+                    };
+
+                    return buffer -> {
+                        try {
+                            if (buffer.isEmpty()) {
+                                return;
+                            }
+                            for (Object item : buffer) {
+                                // on duplicate key, we'll flush immediately
+                                K key = toKeyFn.apply((T) item);
+                                if (tmpMap.containsKey(key)) {
+                                    map.executeOnKeys(tmpMap.keySet(), entryProcessor);
+                                    tmpMap.clear();
+                                }
+                                tmpMap.put(key, (T) item);
+                            }
+                            map.executeOnKeys(tmpMap.keySet(), entryProcessor);
+                            tmpMap.clear();
+                        } catch (HazelcastInstanceNotActiveException e) {
+                            handleInstanceNotActive(instance, e, isLocal);
+                        }
+                        buffer.clear();
+                    };
+                },
+                noopConsumer()
+        ));
+    }
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static <T, K, V> ProcessorMetaSupplier updateMapP(
+            String name,
+            ClientConfig clientConfig,
+            DistributedFunction<T, K> toKeyFn,
+            DistributedFunction<T, EntryProcessor<K, V>> toEntryProcessorFn
+    ) {
+        boolean isLocal = clientConfig == null;
+        return dontParallelize(new HazelcastWriterSupplier<>(
+                serializableConfig(clientConfig),
+                index -> new ArrayMap(),
+                ArrayMap::add,
+                instance -> {
+                    IMap map = instance.getMap(name);
+                    return buffer -> {
+                        try {
+                            if (buffer.isEmpty()) {
+                                return;
+                            }
+                            for (Entry<Object, Object> entry : buffer.entrySet()) {
+                                EntryProcessor<K, V> entryProcessor = toEntryProcessorFn.apply((T) entry);
+                                K key = toKeyFn.apply((T) entry);
+                                map.submitToKey(key, entryProcessor);
```

Comment:
what happens to the result of the future? It doesn't look like it gets handled at all. It will also not work with snapshotting and processing guarantees. It should behave in a similar way to `WriteKafkaP`.